File: Documentation Code Snippets/ScanbotSDK Examples/CheckRecognizer/DetectionOnImage/Objc/CheckDetectionOnImage.m
//
//  CheckDetectionOnImage.m
//  ScanbotSDK Examples
//
//  Created by Rana Sohaib on 27.03.23.
//

#import "CheckDetectionOnImage.h"
@import ScanbotSDK;

@implementation CheckDetectionOnImage

- (void) detectCheckOnImage {
    
    // Image containing Check.
    UIImage *image = [UIImage imageNamed:@"checkImage"];
    
    // Creates an instance of `SBSDKCheckRecognizer`.
    SBSDKCheckRecognizer *detector = [[SBSDKCheckRecognizer alloc] init];
    
    // Type of checks that needs to be detected.
    NSArray<SBSDKCheckDocumentRootType *> *acceptedCheckTypes = [SBSDKCheckDocumentRootType allDocumentTypes];
    
    // Set the types of check on detector.
    [detector setAcceptedCheckTypes:acceptedCheckTypes];
    
    // Returns the result after running detector on the image.
    SBSDKCheckRecognizerResult *result = [detector recognizeOnImage:image];
}

@end

File: Documentation Code Snippets/ScanbotSDK Examples/CheckRecognizer/DetectionOnImage/CheckDetectionOnImage.swift
//
//  CheckDetectionOnImage.swift
//  ScanbotSDK Examples
//
//  Created by Rana Sohaib on 27.03.23.
//

import Foundation
import ScanbotSDK

func recognizeCheckOnImage() {
    
    // The image containing a check.
    guard let image = UIImage(named: "checkImage") else { return }
    
    // Create an instance of `SBSDKCheckScanner`.
    let scanner = SBSDKCheckScanner()
    
    // Set the desired check types to the scanner.
    scanner.acceptedCheckTypes = SBSDKCheckDocumentModelRootType.allDocumentTypes
    
    // Scan the check from given image.
    let result = scanner.scan(from: image)
}
